Allow independent subviews

maxpatiiuk opened the pull request:
Fixes #114
Fixes #3127

but only if sub view is displayed as a button (because button opens a dialog, and dialog has a save button, and dialog creates a separate `<form>` - thus validation errors in independent sub-view are scoped to that sub-view)

This is just a one-line change because 50% of the work was done in 7.7
Now, the other 50% is testing - **please test the hell out of this**

Suggestions for testing:
- Collection Object -> Accession as a subview button
- Collecting Event -> Collection as a subview
   - try opening Collection Object -> Collecting Event -> Collection Object that way
- Save main form before saving sub view
   - Especially when sub view has validation errors
- Also do testing in record sets (especially with saving and validation errors and going between record set records)
- Test #3127

chanulee1 reviewed (changes requested):
Cannot associate CO with an Accession through the Accession form. After adding a CO and reloading, see that the CO has not been added. Querying shows that the relationship has not been established. Also, adding the CO does not trigger the save button.

https://user-images.githubusercontent.com/103299204/224388643-a5c9b70f-d660-4e47-9d33-91acde3c01d8.mp4


https://coldfish-issue-114.test.specifysystems.org/specify/view/accession/827/

grantfitzsimmons commented:
> Cannot associate CO with an Accession through the Accession form. After adding a CO and reloading, see that the CO has not been added. Querying shows that the relationship has not been established. Also, adding the CO does not trigger the save button.

I was able to recreate this issue without using an auto-generated form as well.

grantfitzsimmons reviewed (changes requested):
Same issue as @chanulee1, I cannot add items to an independent subview and have them remain.

maxxxxxdlp commented:
Also, need to add ability to add existing independent subviews using search, rather than create new (like in sp6)

melton-jason committed:
Extend frontend orm to allow handling of independent resources

melton-jason committed:
Allow adding independent resources from Subviews

melton-jason committed:
Remove independentResources type from SpecifyResource

melton-jason committed:
Cleanup imports

melton-jason committed:
Merge remote-tracking branch 'origin/production' into issue-114

melton-jason committed:
Restore non-button subviews for depedent resources

melton-jason committed:
Merge remote-tracking branch 'origin/production' into issue-114

melton-jason committed:
Merge remote-tracking branch 'origin/production' into issue-114

melton-jason committed:
Lint code with ESLint and Prettier

Triggered by b82285b1bed2ed2a61ac88198245f80f65c27396 on branch refs/heads/issue-114

grantfitzsimmons committed:
Merge branch 'production' into issue-114

grantfitzsimmons committed:
Merge branch 'production' into issue-114

melton-jason committed:
Merge branch 'production' into issue-114

melton-jason committed:
Lint code with ESLint and Prettier

Triggered by fc52b1eb81d5be061085e27d4ab50021269021c9 on branch refs/heads/issue-114

melton-jason committed:
Merge branch 'production' into issue-114

grantfitzsimmons committed:
Merge branch 'production' into issue-114
